WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+        }
+
+        const iDynTree::Model &model = kinDynComp.model();
+
+        // Prepare pointer to KinDynComputations object
+        kinDyn = std::make_shared<iDynTree::KinDynComputations>();
+        kinDyn->setFrameVelocityRepresentation(iDynTree::FrameVelocityRepresentation::MIXED_REPRESENTATION);
+        kinDyn->loadRobotModel(mdlLoader.model());
+
+        jointVelocities.setZero(kinDyn->model().getNrOfDOFs());
+        jointPositions.setZero(kinDyn->model().getNrOfDOFs());
+
+        kinDyn->getRobotState(basePose, jointPositions, baseVelocity, jointVelocities, gravity);
+
+        dynamics = std::make_shared<FloatingBaseSystemKinematics>();
+
+        // initial state
+        dynamics->setState({basePose.topRightCorner<3, 1>(),            // base postion
+                            toManifRot(basePose.topLeftCorner<3, 3>()), // base rotation
+                            jointPositions});                           // joint position
+
+        integrator = std::make_shared<ForwardEuler<FloatingBaseSystemKinematics>>();
+        integrator->setIntegrationStep(dT); // dt->period
+        integrator->setDynamicalSystem(dynamics);
+
+        // Prepare the parameter handler. The task setup is done through this. This can be substituted with a configuration file
+        auto parameterHandler = std::make_shared<StdImplementation>();
+
+        auto ikParameterHandler = std::make_shared<StdImplementation>();
+        ikParameterHandler->setParameter("robot_velocity_variable_name", "robotVelocity");
+        ikParameterHandler->setParameter("verbosity", false);
+        parameterHandler->setGroup("IK", ikParameterHandler);
+
+        parameterHandler->setParameter("tasks",
+                                       std::vector<std::string>{"THUMB_DISTANCE_TASK",
+                                                                "INDEX_DISTANCE_TASK",
+                                                                "MIDDLE_DISTANCE_TASK",
+                                                                "THUMB_GRAVITY_TASK",
+                                                                "INDEX_GRAVITY_TASK",
+                                                                "MIDDLE_GRAVITY_TASK",
+                                                                "PALM_GRAVITY_TASK",
+                                                                "SE3_TASK",
+                                                                "REGULARIZATION_TASK",
+                                                                "JOINT_LIMITS_TASK"});
+
+        // Setup for the distance task
+
+        // thumb
+        auto thumbDistanceParameterHandler = std::make_shared<StdImplementation>();
+        thumbDistanceParameterHandler->setParameter("type", "DistanceTask");
+        thumbDistanceParameterHandler->setParameter("kp", convergenceRate);
+        const Eigen::VectorXd thumbWeightRegularization = 1000 * Eigen::VectorXd::Ones(1);
```

Comment:
Maybe this regularization parameter should be defined from file (as the ones below)